heka/_service.sls

```diff
@@ -119,8 +119,6 @@ heka_{{ service_name }}_service:
 
 {# Loading the other services' support metadata for local roles #}
 
-{%- if service_name in ['log_collector', 'metric_collector'] %}
-
 {%- for service_name, service in pillar.iteritems() %}
 {%- if service.get('_support', {}).get('heka', {}).get('enabled', False) %}
```

Comment:
Now that we get all the services' support metadata for the remote_collector and the aggregator as well, you can remove the code section below where we read the Heka support metadata for the remote_collector and the aggregator.